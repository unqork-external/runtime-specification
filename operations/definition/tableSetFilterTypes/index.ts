export * from './tableSetFilterTypes.model'
export * from './tableSetFilterTypes.options'
