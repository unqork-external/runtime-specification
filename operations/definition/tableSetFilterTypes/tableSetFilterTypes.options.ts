import { CollectionOf, Description, Required } from '@tsed/schema'

import { ColumnFilterType } from '../../../components/muiBasicTable/filters/columnFilterType'

export class TableSetFilterTypesOperationOptions {
  @Required()
  @Description('Key of the Table component to set column filter types.')
  targetKey: string

  @Required()
  @Description('The columns of the Table component to set filter types.')
  @CollectionOf(ColumnFilterType)
  filters: ColumnFilterType[]
}
