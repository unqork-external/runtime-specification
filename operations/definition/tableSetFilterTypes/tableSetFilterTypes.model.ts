import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { TableSetFilterTypesOperationOptions } from './tableSetFilterTypes.options'
import { OpsBuilderStable, Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_SET_FILTER_TYPES)
@Description('Set the types and custom options for column filters in the Table component.')
@Stable()
@OpsBuilderStable()
export class TableSetFilterTypesOperation extends Operation<OperationTypes.TABLE_SET_FILTER_TYPES> {
  @Required()
  public type = OperationTypes.TABLE_SET_FILTER_TYPES as const

  @Required()
  public options: TableSetFilterTypesOperationOptions = new TableSetFilterTypesOperationOptions()
}
