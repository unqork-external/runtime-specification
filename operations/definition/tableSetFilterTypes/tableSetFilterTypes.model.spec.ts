import { TableSetFilterTypesOperation } from './tableSetFilterTypes.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableSetFilterTypes Tests', () => {
  it('TableSetFilterTypes Model', () => {
    const { schema } = generateSchemaAndValidate(TableSetFilterTypesOperation)

    expect(schema).toMatchSnapshot()
  })
})
