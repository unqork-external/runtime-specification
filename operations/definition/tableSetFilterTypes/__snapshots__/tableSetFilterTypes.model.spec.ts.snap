// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableSetFilterTypes Tests TableSetFilterTypes Model 1`] = `
{
  "definitions": {
    "ColumnFilterType": {
      "properties": {
        "dateFormat": {
          "description": "The date format shown in the date range filter UI.",
          "type": "string",
        },
        "id": {
          "description": "ID of the column to set filter type.",
          "minLength": 1,
          "type": "string",
        },
        "max": {
          "description": "The maximum value that the number range filter can accept. This is an inclusive, meaning rows with this value will be included.",
          "type": "number",
        },
        "min": {
          "description": "The minimum value that the number range filter can accept. This is an inclusive, meaning rows with this value will be included.",
          "type": "number",
        },
        "step": {
          "description": "The interval between acceptable values in number range filter. Cannot exceed the max value for the range or be 0.",
          "type": "number",
        },
        "type": {
          "description": "Type of filter for the column.",
          "enum": [
            "checkbox",
            "text",
            "select",
            "multi-select",
            "range-slider",
            "range",
            "date-range",
          ],
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "id",
        "type",
      ],
      "type": "object",
    },
    "TableSetFilterTypesOperationOptions": {
      "properties": {
        "filters": {
          "description": "The columns of the Table component to set filter types.",
          "items": {
            "$ref": "#/definitions/ColumnFilterType",
          },
          "type": "array",
        },
        "targetKey": {
          "description": "Key of the Table component to set column filter types.",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "filters",
      ],
      "type": "object",
    },
  },
  "description": "Set the types and custom options for column filters in the Table component.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableSetFilterTypesOperationOptions",
    },
    "type": {
      "const": "TABLE_SET_FILTER_TYPES",
      "examples": [
        "TABLE_SET_FILTER_TYPES",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "stability": "STABLE",
  "type": "object",
}
`;
